templates/includes/pagination.html

```diff
@@ -0,0 +1,15 @@
+<ul class="pagination">
+  {% if page_obj.has_previous %}
+    <li class="page-item">
+      <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">prev</a>
```

Comment:
There is an extra space after the equal sign in the href attribute. It should be `href="?page={{ page_obj.previous_page_number }}"` without the space.